airflow/providers/dbt/cloud/hooks/dbt.py

```diff
@@ -181,33 +181,46 @@ def __init__(self, dbt_cloud_conn_id: str = default_conn_name, *args, **kwargs)
         super().__init__(auth_type=TokenAuth)
         self.dbt_cloud_conn_id = dbt_cloud_conn_id
 
+    @staticmethod
+    def _get_tenant_domain(conn: Connection) -> str:
+        if conn.schema:
+            warnings.warn(
```

Comment:
We can make a major release with the updated behavior rather than carry the deprecation further

Reply:
Done. Do you think this warrants a newsfragment or does the major version increment imply something "newsworthy"?

Reply:
You will need to update version in provider.yaml to 3.0.0 and also update provider changelog with details of the breaking change (check Amazon provider change log for example)

Reply:
Oh right 🤦. I updated these now along with the PR title and description to make it clear what the significant change is in this PR.